libraries/lib-module-manager/PluginHost.cpp

```diff
@@ -35,14 +35,18 @@ namespace
          if(auto provider = ModuleManager::Get().CreateProviderInstance(providerId, wxEmptyString))
          {
             TranslatableString errorMessage{};
-
-            provider->DiscoverPluginsAtPath(pluginPath, errorMessage, [&result](PluginProvider *provider, ComponentInterface *ident)
+            auto validator = provider->MakeValidator();
+            provider->DiscoverPluginsAtPath(pluginPath, errorMessage, [&](PluginProvider *provider, ComponentInterface *ident)
             {
                //Workaround: use DefaultRegistrationCallback to create all descriptors for us
                //and then put a copy into result
                auto id = PluginManager::DefaultRegistrationCallback(provider, ident);
                if(const auto desc = PluginManager::Get().GetPlugin(id))
+               {
+                  if(validator)
+                     validator->Validate(*ident);
```

Comment:
This line is in a necessary catch block, but I would also define Validate to return bool, so that the failure paths in the overrides don't need to throw more exceptions.

This opinion isn't a blocker for this PR.